### ISSUE TITLE
NDCubeSequence Indexing Bug

### ISSUE THREAD

**Issue by DanRyanIrish** [sunpy/ndcube#22]
```
# Imports
from astropy.io import fits
from astropy.wcs import WCS
from ndcube import NDCube, NDCubeSequence

# Define files with which bug was found.
file0 = 'iris_l2_20141211_191222_3803105278_raster_t000_r00000.fits'
file1 = 'iris_l2_20141211_191222_3803105278_raster_t000_r00001.fits'

# Build NDCubes from C II 1336 data
ext_num = 1
hdulist0 = fits.open(file0)
ndc0 = NDCube(hdulist0[ext_num].data, WCS(hdulist0[ext_num].header))
hdulist0.close()
hdulist1 = fits.open(file1)
ndc1 = NDCube(hdulist1[ext_num].data, WCS(hdulist1[ext_num].header))
hdulist1.close()

# Build NDCubeSequence
nds = NDCubeSequence([ndc0, ndc1], common_axis=0)
```
```nds.index_as_cube[2]``` errors with the following traceback.
```
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-56-f3fe4c7f3673> in <module>()
----> 1 nds.index_as_cube[2]

~/sunpy_dev/ndcube/ndcube/ndcube.py in __getitem__(self, item)
    695 
    696     def __getitem__(self, item):
--> 697         return cube_utils.index_sequence_as_cube(self.seq, item)
    698 
    699 

~/sunpy_dev/ndcube/ndcube/cube_utils.py in index_sequence_as_cube(cubesequence, item)
    183     # (int, rest of the information) not (int, int , rest of the information)
    184     if isinstance(item_tuple[0], int) and isinstance(item_tuple[1], int):
--> 185         return get_cube_from_sequence(cubesequence, item_tuple[1::])
    186     else:
    187         return get_cube_from_sequence(cubesequence, item_tuple, type_slice='sequence_as_cube')

~/sunpy_dev/ndcube/ndcube/cube_utils.py in get_cube_from_sequence(cubesequence, item, type_slice)
     61                 result = result.data[item[0]][item[1]]
     62             else:
---> 63                 result = result.data[item[0]][item[1::]]
     64         # the 0th index of tuple will have the slice that will be applied across cubes
     65         # the 1st index of tuple contains the information of 1st cube to slice and last

IndexError: list index out of range
```

**Comment by DanRyanIrish** [comment]
@abit2, here is the example I was talking about.

**Comment by DanRyanIrish** [comment]
Having read through the code based associated with this error, it occurs to me that perhaps we need an ```IndexAsCubeItem``` class (not sure about the name) that derives the information of the slices across the different cubes from the input slicing item.  That way the different slicing functions can always assume the same type of input.  This would make understanding what is happening in each case in the code much easier to understand.

What do you think @abit2?

**Comment by abit2** [comment]
Yes I think it should be item_tuple not item_tuple[1::].

I made a mistake there, I was indexing the seq according to the cube and applying all other slices but that is wrong as the index_as_cube sees the whole sequence as one complete cube.

**Comment by abit2** [comment]
Yes I agree code needs to simplified. It takes time to debug, even I have a tough time. :)

**Comment by DanRyanIrish** [comment]
@abit2, or is one of the issues here that the ```type_slice='sequence_as_cube'``` is not needed?  From looking at the ```cube_utils.get_cube_from_sequence()```, it seems ```type_slice``` is only checked when ```isinstance(item[0], slice)```.  But there is no case for when ```isinstance(item[0], slice) and type_slice == None```.  So doesn't that mean that it if ```type_slice``` was never checked, the code would still do the same thing?

This would mean that we could get rid of the ```if...else...``` statement in ```index_sequence_as_cube``` (https://github.com/sunpy/ndcube/blob/master/ndcube/cube_utils.py#L184) and replace it with a single line, ```return get_cube_from_sequence(cubesequence, item_tuple)```.

**Comment by DanRyanIrish** [comment]
P.S. Well if you have a hard time debugging it then it's definitely complicated ;)

**Comment by DanRyanIrish** [comment]
Addressed by PR #23
